File: LeetCode Solutions/March 2021 LeetCoding Challenge/Keys and Rooms/Keys and Rooms.cpp
class Solution {
public:
    bool canVisitAllRooms(vector<vector<int>>& rooms) {
        vector<bool> arr(rooms.size(), false);
        arr[0] = true;
        stack<int> s = stack<int>({0});
        int count = 1;
        while (s.size()){
            vector<int> keys = rooms[s.top()];
            s.pop();
            for (int k : keys){
                if (!arr[k]){
                    s.push(k), arr[k] = true, count++;
                }
            }
        }
        return rooms.size() == count;
    }
};

File: LeetCode Solutions/March 2021 LeetCoding Challenge/Keys and Rooms/Keys and Rooms.py
class Solution:
    def canVisitAllRooms(self, rooms: List[List[int]]) -> bool:
        arr, stack, count = [False for i in range(len(rooms))], [0], 1
        arr[0] = 1
        while (stack):
            keys = rooms[stack.pop()]
            for i in keys:
                if (not arr[i]):
                    stack.append(i)
                    arr[i] = True
                    count += 1
        return len(rooms) == count